### ISSUE TITLE
Support type alias groups

### ISSUE THREAD

**Issue by matthewmueller** [golang/lint#439]
First of all, `golint` is an indispensable that makes coding so much better. Thanks for putting this together!

Today I noticed that golint is a bit inconsistent with out it treats comment groups. I've attached an example:

<img width="354" alt="screen shot 2019-03-05 at 12 49 57 pm" src="https://user-images.githubusercontent.com/170299/53783767-8f3ddc00-3f45-11e9-9f81-f19e40af2afa.png">

You'll notice that type aliases cannot have a top-level comment though this is valid Go code. Here's the output from `golint`

```sh
internal/stripe/stripe.go:59:1: comment on exported type Customer should be of the form "Customer ..." (with optional leading article)
internal/stripe/stripe.go:59:1: comment on exported type CustomerParams should be of the form "CustomerParams ..." (with optional leading article)
```

Definitely not a dealbreaker, but it'd be good to apply the same rule to type alias groups as well.

**Comment by mvdan** [comment]
Thank you for submitting this issue! As per https://github.com/golang/go/issues/38968, we are freezing and deprecating golint. There's no drop-in replacement to golint per se, but you should find that [Staticcheck](https://staticcheck.io/) works well in encouraging good Go code, much like golint did in the past, since it also includes style checks. There's always `gofmt` and `go vet` too, of course.

If you would like to contribute further, I'd encourage you to engage [Staticcheck's issue tracker](https://github.com/dominikh/go-tools/issues) or look at [vet's open issues](https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+cmd%2Fvet+in%3Atitle), as they are both actively maintained. If you have an idea that doesn't fit into either of those tools, you could look at [other Go linters](https://github.com/golangci/awesome-go-linters), or write your own - these days it's fairly straightforward with [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis).

To help avoid confusion, I'm closing all issues before we freeze the repository. If you have any feedback, you can leave a comment on the proposal thread where it was decided to deprecate golint - though note that the proposal has been accepted for nearly a year. Thanks!
